### PR TITLE
Compatibility with JDK 13 and 14

### PR DESCRIPTION
**Pull request by tomparle**
# Pull Request Checklist

* [ ] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [ ] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [ ] Have you referenced any issues you're fixing using [commit message keywords](https://help.github.com/articles/closing-issues-using-keywords/)?
* [ ] Have you updated the documentation?
* [ ] Have you added tests for any changed functionality?

# Helpful things

## Fixes

Fixes https://github.com/playframework/play1/issues/1324

## Purpose

Make Play apps runnable with JDK version up to 14.
See corresponding issue for details and migration steps for faster-gt module users.

### CODE REVIEW

File: framework/src/play/classloading/ApplicationCompiler.java [hunk 1]
```diff
@@ -37,14 +37,13 @@
...
-    	compatibleJavaVersions.put("9", CompilerOptions.VERSION_9);
-    	compatibleJavaVersions.put("10", CompilerOptions.VERSION_10);
-    	compatibleJavaVersions.put("11", CompilerOptions.VERSION_11);
-	}
+    static final Map<String, String> compatibleJavaVersions = Map.of( //
+    		"1.8", CompilerOptions.VERSION_1_8, //
+    		"9", CompilerOptions.VERSION_9, //
+    		"10", CompilerOptions.VERSION_10, //
+    		"11", CompilerOptions.VERSION_11, //
+    		"12", CompilerOptions.VERSION_12 //
```
Comment (asolntsev) [inline framework/src/play/classloading/ApplicationCompiler.java:45]: Shouldn't we add "13" and "14" versions?

Comment (tomparle) [reply framework/src/play/classloading/ApplicationCompiler.java:45]: Option "13" is not available with this version of the JDT compiler (see previous remarks).
Option "14" is not available even in the latest version of the lib.

**Comment by tomparle** [comment]
Sorry I used a method available only from JDK 9, which made the build fail. I'll fix this tomorrow.

**Comment by tomparle** [comment]
(build fixed)

File: framework/src/play/classloading/ApplicationCompiler.java (hunk 1 above)
Comment (xael-fry) [reply framework/src/play/classloading/ApplicationCompiler.java:45]: it is already dine in #1338

**Review by xael-fry — COMMENTED** [review]
look good to me

**Comment by xael-fry** [comment]
@tomparle can you resolve the conflicts?

**Pull request merged**